pilot/pkg/model/push_context.go

```diff
@@ -746,11 +746,14 @@ func (ps *PushContext) GatewayServices(proxy *Proxy) []*Service {
 			}
 
 			for _, host := range virtualServiceDestinationHosts(vs) {
-				hostsFromGateways[host] = struct{}{}
+				hostsFromGateways.Insert(host)
 			}
 		}
 	}
 
+	hostsFromMeshConfig := ps.getHostsFromMeshConfig(proxy)
+	hostsFromGateways.Insert(hostsFromMeshConfig.UnsortedList()...)
```

Comment:
performance nit: its wasteful to convert to list just to add it. We could add some new method `func (s Set) Merge(other Set)`, which is like Union but it is in place...?

Reply:
sounds good, will update the implements.

Reply:
updated